clients/drcachesim/tracer/instru_offline.cpp

```diff
@@ -403,17 +403,23 @@ int
 offline_instru_t::append_unit_header(byte *buf_ptr, thread_id_t tid, ptr_int_t window)
 {
     byte *new_buf = buf_ptr;
-    offline_entry_t *entry = (offline_entry_t *)new_buf;
-    entry->timestamp.type = OFFLINE_TYPE_TIMESTAMP;
-    uint64 frozen = frozen_timestamp_.load(std::memory_order_acquire);
-    entry->timestamp.usec = frozen != 0 ? frozen : instru_t::get_timestamp();
-    new_buf += sizeof(*entry);
+    new_buf += append_timestamp(new_buf);
```

Comment:
Related note: can we add an invariant check that verifies that timestamps are monotonically increasing?

Reply:
@ivankyluk already added that, and in fact I hit it earlier in this PR and had to fix by changing the split pipe header behavior: see the PR description and [1bcb72e](https://github.com/DynamoRIO/dynamorio/pull/6290/commits/1bcb72e422c7b82f63cefb228c23c53dc0c435ca).